src/20240603_Multi_MayFundingUpdate/AaveV3Ethereum_MayFundingUpdate_20240603.sol

```diff
@@ -0,0 +1,319 @@
+  using SafeERC20 for IERC20;
+
+  AaveSwapper public constant SWAPPER = AaveSwapper(MiscEthereum.AAVE_SWAPPER);
+  IAaveWstethWithdrawer public constant AAVE_STETH_WITHDRAWER =
+    IAaveWstethWithdrawer(0x2C4d3C146b002079949d7EecD07f261A39c98c4d);
+
+  address public constant GHO_USD_FEED = 0x3f12643D3f6f874d39C2a4c9f2Cd6f2DbAC877FC;
+  address public constant MILKMAN = 0x11C76AD590ABDFFCD980afEC9ad951B160F02797;
+  address public constant PRICE_CHECKER = 0xe80a1C615F75AFF7Ed8F08c9F21f9d00982D666c;
+  address public constant FRONTIER_SAFE = 0xCDb4fA6ba08bF1FB7Aa9fDf6002E78EDc431a642;
+  uint256 public constant WSTETH_AMOUNT = 350 ether;
+  uint256 public constant FRONTIER_ALLOWANCE_AMOUNT = 2493 ether;
+  uint256 public constant AETH_USDT_AMOUNT = 2_000_000e6;
+  uint256 public constant AETH_USDC_AMOUNT = 1_500_000e6;
+
+  function execute() external {
+    _withdrawAndSwapForGHO();
+    _migrateWeth();
+    _depositAllLink();
+    _unwrapLsts();
+
+    /// FRONTIER allowance
+    AaveV3Ethereum.COLLECTOR.approve(
+      AaveV3EthereumAssets.WETH_UNDERLYING,
+      FRONTIER_SAFE,
+      FRONTIER_ALLOWANCE_AMOUNT
+    );
+  }
+
+  function _withdrawAndSwapForGHO() internal {
+    /// aEthDAI & aDAI
+    AaveV3Ethereum.COLLECTOR.transfer(
+      AaveV3EthereumAssets.DAI_A_TOKEN,
+      address(this),
+      IERC20(AaveV3EthereumAssets.DAI_A_TOKEN).balanceOf(address(AaveV3Ethereum.COLLECTOR)) - 1e18
+    );
+
+    AaveV3Ethereum.COLLECTOR.transfer(
+      AaveV2EthereumAssets.DAI_A_TOKEN,
+      address(this),
+      IERC20(AaveV2EthereumAssets.DAI_A_TOKEN).balanceOf(address(AaveV3Ethereum.COLLECTOR)) - 1e18
+    );
+
+    AaveV3Ethereum.POOL.withdraw(
+      AaveV3EthereumAssets.DAI_UNDERLYING,
+      type(uint256).max,
+      address(SWAPPER)
+    );
+
+    AaveV2Ethereum.POOL.withdraw(
+      AaveV2EthereumAssets.DAI_UNDERLYING,
+      type(uint256).max,
+      address(SWAPPER)
+    );
+
+    uint256 daiBalance = IERC20(AaveV3EthereumAssets.DAI_UNDERLYING).balanceOf(address(SWAPPER));
+
+    SWAPPER.swap(
+      MILKMAN,
+      PRICE_CHECKER,
+      AaveV3EthereumAssets.DAI_UNDERLYING,
+      AaveV3EthereumAssets.GHO_UNDERLYING,
+      AaveV3EthereumAssets.DAI_ORACLE,
+      GHO_USD_FEED,
+      address(AaveV3Ethereum.COLLECTOR),
+      daiBalance,
+      50
+    );
+
+    /// aEthLUSD
+    AaveV3Ethereum.COLLECTOR.transfer(
+      AaveV3EthereumAssets.LUSD_A_TOKEN,
+      address(this),
+      IERC20(AaveV3EthereumAssets.LUSD_A_TOKEN).balanceOf(address(AaveV3Ethereum.COLLECTOR)) - 1e18
+    );
+
+    AaveV3Ethereum.POOL.withdraw(
+      AaveV3EthereumAssets.LUSD_UNDERLYING,
+      type(uint256).max,
+      address(SWAPPER)
+    );
+
+    uint256 lusdBalance = IERC20(AaveV3EthereumAssets.LUSD_UNDERLYING).balanceOf(address(SWAPPER));
+
+    SWAPPER.swap(
+      MILKMAN,
+      PRICE_CHECKER,
+      AaveV3EthereumAssets.LUSD_UNDERLYING,
+      AaveV3EthereumAssets.GHO_UNDERLYING,
+      AaveV3EthereumAssets.LUSD_ORACLE,
```

Comment:
@defijesus The LUSD feed being used is the capo feed, I would use the chainlink feed